docs/contracts.rst

```diff
@@ -908,6 +908,33 @@ Such contracts cannot be compiled (even if they contain implemented functions al
 
 If a contract inherits from an abstract contract and does not implement all non-implemented functions by overriding, it will itself be abstract.
 
+.. index:: ! contract;interface, ! interface contract
+
+**********
+Interfaces
+**********
+
+Interfaces are similar to abstract contracts, but they cannot have any functions implemented. There are further restrictions:
+
+#. Cannot inherit other contracts or interfaces.
+#. Cannot define variables.
+#. Cannot define structs.
+
+(Structs might be supported in the future if the ABI adds support to them.)
+
+Interfaces are basically limited to what the Contract ABI can represent and the conversion between the ABI and
+an Interface should be possible without any information loss.
```

Comment:
How about adding: "One exception is types not supported by ABI, but convertible to an ABI primitive possible in interfaces. An example is a named contract can be referenced in an interface, while it is represented as an ``address`` in the ABI."